Scripts/RemediationScripts/Remediate-DisableLocalAccountsForKubernetesService.ps1

```diff
@@ -0,0 +1,867 @@
+            $aadClientIds = @()
+            $res = @()
+
+            Write-Host "Disabling local accounts for Kubernetes cluster [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            # Check whether RBAC is enabled on cluster or not. Without RBAC, AAD cannot be enabled.
+            if ($isRBACEnabled -eq $true)
+            {
+                
+                if($isAADEnabled)
+                {
+                    $res = $disableLocalAccounts.DisableLocalAccountsForKuberneteCluster($subscriptionId, $resourceName, $resourceGroupName, $location, $true, $aadClientIds)
+                }
+                else 
+                {
+                    if ($addAADGroup)
+                    {
+                        $userSkippedAdminGroups = $false
+                        do
+                        {
+                            Write-Host "Please provide Azure AD group object id (group will be registered as an admin group on the cluster): "
+                            $aadClientIds = Read-Host
+                            Write-Host $([Constants]::SingleDashLine)
+
+                            if ([String]::IsNullOrWhiteSpace($aadClientIds))
+                            {
+                                Write-Host "Given object id is empty."
+                                Write-Host "Do you want to re-enter Azure AD group object id? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+
+                                $userInput = Read-Host -Prompt "(Y|N)"
+                                Write-Host $([Constants]::SingleDashLine)
+                                if($userInput -ne "Y")
+                                {
+                                    $addAADGroup = $false
+                                    $userSkippedAdminGroups = $true
+                                    Write-Host "Azure AD group will not be added for Kubernetes cluster [$($resourceName)]." -ForegroundColor $([Constants]::MessageType.Update)
+                                    Write-Host $([Constants]::SingleDashLine)
+                                }
+                            }
+                        }
+                        While ([String]::IsNullOrWhiteSpace($aadClientIds) -and (-not $userSkippedAdminGroups))
+                    }
+                    else 
+                    {
+                        Write-Host "AAD group will not be added to Kubernetes cluster [$($resourceName)]." -ForegroundColor $([Constants]::MessageType.Update)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                                
+                    # Setting local accounts profile config to Kubernetes cluster.
+                    $res = $disableLocalAccounts.DisableLocalAccountsForKuberneteCluster($subscriptionId, $resourceName, $resourceGroupName, $location, $false, $aadClientIds)                 
```

Comment:
what will be the response if addClientIds passed are null?

Reply:
It will configure AAD for Kubernetes but will not add any particular group.